terraform-provider/internal/service/subgraph/data_source_cosmo_subgraph.go

```diff
@@ -0,0 +1,169 @@
+package subgraph
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/wundergraph/cosmo/terraform-provider-cosmo/internal/api"
+	"github.com/wundergraph/cosmo/terraform-provider-cosmo/internal/client"
+	"github.com/wundergraph/cosmo/terraform-provider-cosmo/internal/utils"
+)
+
+var _ datasource.DataSource = &SubgraphDataSource{}
+
+func NewSubgraphDataSource() datasource.DataSource {
+	return &SubgraphDataSource{}
+}
+
+type SubgraphDataSource struct {
+	*client.PlatformClient
+}
+
+type SubgraphDataSourceModel struct {
+	Id                   types.String `tfsdk:"id"`
+	Name                 types.String `tfsdk:"name"`
+	Namespace            types.String `tfsdk:"namespace"`
+	RoutingUrl           types.String `tfsdk:"routing_url"`
+	BaseSubgraphName     types.String `tfsdk:"base_subgraph_name"`
+	Labels               types.List   `tfsdk:"labels"`
+	SubscriptionUrl      types.String `tfsdk:"subscription_url"`
+	SubscriptionProtocol types.String `tfsdk:"subscription_protocol"`
+	Readme               types.String `tfsdk:"readme"`
+	WebsocketSubprotocol types.String `tfsdk:"websocket_subprotocol"`
+	IsEventDrivenGraph   types.Bool   `tfsdk:"is_event_driven_graph"`
+	IsFeatureSubgraph    types.Bool   `tfsdk:"is_feature_subgraph"`
+	Headers              types.List   `tfsdk:"headers"`
```

Comment:
When do we use headers?

Reply:
We use headers on Update Requests in Subgraphs. This comes from the generated proto buf resources.

Seems like headers can't be passed on creation, but can on update of the subgraph.

Reply:
Yes, that confused me. We never store headers.

Reply:
Might be obsolete:

https://github.com/wundergraph/cosmo/blob/326751f6162c80522a66f81f6d7b1c9be2104ce9/proto/wg/cosmo/platform/v1/platform.proto#L537

I will hide it in the provider.